samples/spring-transfer-workflow/src/main/java/com/example/transfer/TransferState.java

```diff
@@ -0,0 +1,43 @@
+package com.example.transfer;
+
+import static com.example.transfer.TransferState.TransferStatus.*;
+
+// tag::domain[]
+public record TransferState(Transfer transfer, TransferStatus status) {
+
+  public record Transfer(String from, String to, int amount) { // <1>
+  }
+
+  public enum TransferStatus { // <2>
+    STARTED, MANUAL_APPROVAL_REQUIRED, ACCEPTED, SUCCESSFUL_WITHDRAWAL, COMPLETED, REJECTED
+  }
+  // end::domain[]
+
+  public TransferState(Transfer transfer) {
+    this(transfer, STARTED);
+  }
+
+  public TransferState asAccepted() {
+    return new TransferState(transfer, ACCEPTED);
```

Comment:
For documentation page, we can make it more explicit, ie: `withStatus(ACCEPTED)`. 

Then we can show the `withStatus` method. If we have one specific method each time, we won't show all the methods impl because that's too long. 

When I see the following in the docs
```java
TransferState successfulWithdrawal = transferState.asSuccessfulWithdrawal();
```
I don't know what `asSuccessfulWithdrawal` does unless I have seen the impl for `asSuccessfulWithdrawal` before.

While if I have seen the `withStatus` impl before and later I see

```java
TransferState successfulWithdrawal = transferState.withStatus(SUCCESSFUL_WITHDRAWAL);
```

it becomes immediately clear what is happening.